### DISCUSSION TITLE
Building for iOS: Found architecture 'x86_64', required architecture 'arm64'

### DISCUSSION THREAD

**Discussion by JGKle** [AvaloniaUI/Avalonia#14294] [Q&A] [answered]
My new Avalonia app runs great on Desktop & on Android. I finally just setup a MacInCloud instance to build for iOS - it gets most of the way through, but then fails with:

```
2>clang: error: linker command failed with exit code 1 (use -v to see invocation)
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : clang++ exited with code 1:
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : ld: warning: ignoring duplicate libraries: '-lSystem.IO.Compression.Native', '-lSystem.Native', '-lSystem.Net.Security.Native', '-lSystem.Security.Cryptography.Native.Apple', '-lcompression', '-lmono-component-debugger', '-lmono-component-diagnostics_tracing', '-lmono-component-hot_reload', '-lmonosgen-2.0', '-lxamarin-dotnet-debug'
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : ld: warning: ignoring file '/Users/user935344/Library/Caches/Xamarin/mtbs/builds/MyApp.iOS/e7e514c4285d9b6c2458effc96d3233b51500a4548d9b048f7727c070d38bffd/obj/Debug/net7.0-ios/iossimulator-x64/nativelibraries/libSystem.IO.Compression.Native.dylib': found architecture 'x86_64', required architecture 'arm64'
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : ld: warning: ignoring file '/Users/user935344/Library/Caches/Xamarin/mtbs/builds/MyApp.iOS/e7e514c4285d9b6c2458effc96d3233b51500a4548d9b048f7727c070d38bffd/obj/Debug/net7.0-ios/iossimulator-x64/nativelibraries/libSystem.Native.dylib': found architecture 'x86_64', required architecture 'arm64'
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : ld: warning: ignoring file '/Users/user935344/Library/Caches/Xamarin/mtbs/builds/R
2>Done building project "MyApp.iOS.csproj" -- FAILED.
```

I know this isn't directly an Avalonia issue, but if anyone has any experience with something like this, I'd be appreciative - pretty stuck & unsure how to proceed. I've tried creating a fresh Avalonia project, and I am able to get that to build & run in the remote simulator, so I know the environment is setup correctly. I've tried Googling for `found architecture 'x86_64', required architecture 'arm64'`, but most of what I've seen are solutions for when developing directly in XCode.

Note: the log does actually get cut off like that (i.e. the last line before "Done building project" is truncated).

Thanks in advance

**Comment by maxkatz6** [comment, marked as answer]
"x86_64" is used for simulator. "arm64" is used for the physical device builds.
It all should depend on RuntimeIdentifier (either "ios-arm64" or "iossimulator-x64".
If that helps with diagnosting.

**Reply by JGKle** [reply]
Thanks for the quick reply!

Interesting - so when I opened the iOS csproj for the newly generated Avalonia solution (which worked), there was no RuntimeIdentifier element at all. My project had:

```
    <RuntimeIdentifier>iossimulator-x64</RuntimeIdentifier>
    
    <!-- These properties need to be set in order to run on a real iDevice -->
    <!--<RuntimeIdentifier>ios-arm64</RuntimeIdentifier>-->
    <!--<CodesignKey></CodesignKey>-->
```
...Which makes it seem like it should've been able to run on the simulator, since it was set to iossimulator - but in any event, I just deleted that element, and now it works. Wonder how those tags go there in my project, but looks like I'm good - thanks again!

**Reply by maxkatz6** [reply]
You have pretty old templates, if you had these lines there.

**Reply by JGKle** [reply]
2.5 months? The project was created on 11/11/2023.

**Reply by maxkatz6** [reply]
7 month how these lines were removed
https://github.com/AvaloniaUI/avalonia-dotnet-templates/commit/068dd91c6d8226166a08122bc3dc4d1c888c0f7b

Template itself probably wasn’t re-installed or updated since then

**Reply by JGKle** [reply]
Gotcha. In any case, thanks for the replies - glad it was an easy fix!
